pom.xml

```diff
@@ -185,6 +185,12 @@
             <artifactId>snakeyaml</artifactId>
             <version>${snakeyaml.version}</version>
         </dependency>
+        <dependency>
+            <groupId>javax.management.j2ee</groupId>
+            <artifactId>javax.management.j2ee-api</artifactId>
+            <version>1.1.2</version>
```

Comment:
Note we can't have this as a direct runtime dependency, because normally these classes would be expected to be provided by the J2EE container and should therefore be `<scope>provided</scope>`.

However we also cannot know if these classes will even be available from the class-loader hosting JMXFetch, because when embedded inside `dd-trace-java` it will be deliberately isolated from the application classes. So we should really be using reflection / method handles to peek into the J2EE statistics.

Reply:
Thank you. I changed it accordingly and I also had to create a weak cache for the class and methodhandles since the application can have different mbean connections created in different classloaders so class visibility is not global

Reply:
@amarziali this dependency can be removed now, right?

Reply:
it should be in scope `test`. I thought it was already changed but I forgot it. Thank you